be/src/exec/olap_scan_node.cpp

```diff
@@ -95,7 +95,8 @@ Status OlapScanNode::init(const TPlanNode& tnode, RuntimeState* state) {
 
         _runtime_filter_ctxs[i].runtimefilter = runtime_filter;
     }
-
+    _batch_size = _limit == -1 ? state->batch_size()
```

Comment:
Is this only effective on non-vectorized engine?

Reply:
on vec-exec has the same stmt already